Implementation Sign up

Kouvahe81 opened the pull request:
Cette fonctionnalité permet de créer un utilisateur via un formulaire.
Différents test sont fait à savoir:
- la validité du mot de passe
- la vérification du mot de passe et celle de confirmation
- la force du mot de passe

Kouvahe81 committed:
Implementation Sign up

Kouvahe81 committed:
Implementation Sign up

Kouvahe81 committed:
Implementation Sign up

Kouvahe81 committed:
Implementation Sign up

Kouvahe81 committed:
feature/US7-2/Signup-Unit-test

Kouvahe81 committed:
feature/US7-2/Signup-Unit-test

Kouvahe81 committed:
feature/US7-2/Signup-Unit-test

Kouvahe81 committed:
feature/US7-2/Signup-Unit-test

Julien00859 commented:
Don't commit node_modules! Reset your commits, add node_modules inside your .gitignore and commit again. You can commit the lockfile for CI but not node_modules itself.

https://stackoverflow.com/questions/11459475/should-i-check-in-folder-node-modules-to-git-when-creating-a-node-js-app-on-he

Kouvahe81 committed:
fix: supression du node_modules superflu.

Kouvahe81 committed:
fix: supression du node_modules superflu.

Kouvahe81 committed:
fix: supression du node_modules superflu.

Kouvahe81 committed:
fix: supression du node_modules superflu.

Martin-Hayot commented on g-sonde/src/pages/User/Signup.tsx:
```diff
@@ -0,0 +1,222 @@
+
+import axios from "axios";
+
+
+const SignUp: React.FC = () => {
+	const [firstname, setFirstname] = useState("");
+	const [lastname, setLastname] = useState("");
+    const [email, setEmail] = useState("");
+	const [password, setPassword] = useState(""); 
+    const [confirmPassword, setConfirmPassword] = useState("");            
+	const [toastMessage, setToastMessage] = useState("");
+	const [showErrorToast, setShowErrorToast] = useState(false);
+	const [showSuccessToast, setShowSuccessToast] = useState(false);
+    const [mailValide, setMailValide] = useState(false);
+    const [mdpValide, setMdpValide] = useState(false);
+    const [checkMail, setCheckMail] = useState(false);
+    const [backgroundColor, setBackgroundColor] = useState("");
+    
```
Je pense que ici tu pourrais plutôt utiliser un objet pour contenir tes states plutot que d'allouer un state par valeur

cela pourrait t'aider : https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

Kouvahe81 replied:
Je lirai le lien et amaliorai mon code. Merci

JarodEphec reviewed (commented):
Les fichiers à la racine: yarn.lock, package.json et package-lock.json devraient être supprimés.

JarodEphec commented:
Je vois que tu utilises des fichiers CSS, dans un souci de régularité, pourrais-tu transformer ces fichiers CSS en fonction Tailwind, merci :)

Kouvahe81 committed:
fix: Suppression des fichiers inutiles et utilisation TailwindCSS

Kouvahe81 committed:
fix: Suppression des fichiers inutiles et utilisation TailwindCSS

Kouvahe81 committed:
fix: Suppression des fichiers inutiles et utilisation TailwindCSS

Kouvahe81 commented:
Les modifications ont été faites

Kouvahe81 committed:
fix:correction de quelques erreurs

Kouvahe81 committed:
fix:correction de quelques erreurs

Kouvahe81 committed:
fix:correction de quelques erreurs

Julien00859 commented:
node_modules is still commited, it should not

Kouvahe81 committed:
Delete node_modules directory

Kouvahe81 committed:
Delete node_modules directory

Martin-Hayot committed:
fix: suppression de fichier dans le root du projet

package-lock.json
package.json
yarn.lock

Ces fichiers ne devraient pas se trouver dans le root du projet.

Martin-Hayot committed:
fix: suppression de fichier dans le root du projet

package-lock.json
package.json
yarn.lock

Ces fichiers ne devraient pas se trouver dans le root du projet.

Kouvahe81 committed:
Merge branch 'feature/US7-2/Signup-Create-new-user' of https://github.com/Martin-Hayot/Projet-Integration-2023 into feature/US7-2/Signup-Create-new-user

Samaze123 committed:
# **_US09 :_**

- Ajout du schéma _.sch_ de l'alimentation externe pour la sonde.
- Ajout d'un screenshot _.png_ du circuit pour les personnes ne possédant pas de client permettant de visionner le circuit.

_Le fichier **.sch** doit s'ouvrir dans Eagle ou Fusion 360._

Kouvahe81 committed:
fix: corrections et adaptation

Kouvahe81 committed:
fix: corrections et adaptation

Kouvahe81 committed:
Delete node_modules directory

Kouvahe81 committed:
Delete package.json

Kouvahe81 committed:
Delete package-lock.json

Kouvahe81 committed:
Delete yarn.lock

Kouvahe81 committed:
fix: supression fichiers et déplacement fichiers

Kouvahe81 committed:
fix: supression fichiers et déplacement fichiers

Kouvahe81 committed:
Merge branch 'feature/US7-2/Signup-Create-new-user' of https://github.com/Martin-Hayot/Projet-Integration-2023 into feature/US7-2/Signup-Create-new-user

Kouvahe81 committed:
Update Signup.tsx

Kouvahe81 committed:
chore: déplacement du fichier Signup

Kouvahe81 committed:
chore: déplacement du fichier Signup

Kouvahe81 committed:
Merge branch 'feature/US7-2/Signup-Create-new-user' of https://github.com/Martin-Hayot/Projet-Integration-2023 into feature/US7-2/Signup-Create-new-user

Martin-Hayot reviewed (approved):
Nice ! merci pour toute les modifications suite à mes commentaires.

Kouvahe81 committed:
chore: Résolution des conflits

Kouvahe81 committed:
fix: corrections

JarodEphec committed:
US 11 feature: Ajout des routes restfull api, pour les aquariums.

Cette US a pour but de créer les routes API et les schémas de DB qui permettent :

- La création et la lecture des aquariums dans la DB pour un utilisateur enregistré.
- La création et la lecture des diagnostics dans la DB pour un aquarium.
- L'enregistrement et la lecture des mesures effectuées par une sonde dans la DB pour un aquarium.

Toutes ces routes requièrent l'authentification de l'utilisateur sur base du modèle créé avec l'US 07.

Elle est surement la première itération de l'US 11, cette US va peut-être évoluer en fonction des besoins et de la façon dont les données vont être envoyées à la DB. Description de cette US peut être trouvée [ici](https://github.com/users/Martin-Hayot/projects/2?pane=issue&itemId=40037408).

# Utilisation de l'API

Voici un exemple de l'utilisation de l'API.

## Aquarium

### POST

URL :
```
/api/aquarium
```
Body :
```
{
    name: "Mon aquarium de 2m²",
}
```

### GET

URL :
```
/api/aquarium
```
headers :
```
headers: {
  cookies : [
    accessToken: accessToken,
    refreshToken: refreshToken
  ]
}
```

Réponse :
```
{
    "_id":"6543a8f911cd8288c7143d02",
    "userId":"6533f52718ac173412f5cb86",
    "name":"Mon aquarium de 2m²",
}
```
## Diagnostic

### POST

URL :
```
/api/aquarium/diagnostic
```
Body :
```
{
    aquariumId: "6543a8f911cd8288c7143d02",
    date: "2023-11-02T15:36:48.384Z",
}
```

### GET

URL :
```
/api/aquarium/:aquariumId/diagnostic
```
Parameter :
```
/api/aquarium/654f71941fdc527c68052cc0/diagnostic
```

Réponse :
```
{
    "_id": "6543b6d060921eeaaea1f833",
    "aquariumId": "6543a8f911cd8288c7143d02",
    "date": "2023-11-02T15:36:48.384+00:00",
}
```

## Data

### POST

URL :
```
/api/aquarium/diagnostic/data
```
Body :
```
{
    diagnosticId: 'hfoia97bsocvo1629hfdqsvf',
    measure: 1200,
    frequency: 200,
}
```

### GET

URL :
```
/api/aquarium/:aquariumId/diagnostic/:diagnosticId/data
```
Parameter :
```
/api/aquarium/654f71941fdc527c68052cc0/diagnostic/654f7ad36941701120cea607/data
```

Réponse :
```
{
    "_id": "6543c215181a47e33cadb4fb",
    "diagnosticId": "6543b6d060921eeaaea1f833",
    "measure": 51,
    "frequency": 200,
}
```

bahuwart committed:
fix: correction du bug d'affichage lors du déplacement entre les pages

Samaze123 committed:
US17

US perso de Samuel Lambert.
Creation de la page contact. Cette page est necessaire pour que l'utilisateur qui le souhaite puisse prendre contact avec l'equipe de la sonde.
Lors du contact avec l'equipe, le message de l'utilisateur est sauve dans un ticket.
Pour que l'equipe puisse lire et manager ces tickets, il a fallu creer une page, uniquement pour l'equipe.
Ensuite, pour plus de classification de ces tickets j'ai cree les categories qui sont manager exclusivement par l'equipe mais qui sont utilisable par l'utilisateur.

BACKEND

Modifications Backend

    Routes

    - index.ts : Ajout de toutes les routes dans ce fichier pour facilite l'import
    user.js : Ajout d'une route pour recevoir l'ability d'un utilisateur.
    - auth.js : Ajout d'une reponse avec des donnees de l'utilisateur qui seront stockee dans le localstorage du front. Ajout de ability lors de la creation d'un compte.

    - category.js : Creation de la route pour ajouter, supprimer et recevoir les categories.
      - Il y a aussi une route qui sert a recevoir seulement une categorie grace a son ID.

    - ticket.js : Creation de la route pour ajouter, supprimer et recevoir les tickets.
      - Utilisation du middleware RequireAdmin et requireUser utilise pour certaines url.
      - Details lors du retour si mauvaises donnees entrees.
      - Il y a aussi une route qui sert a recevoir seulement un ticket grace a son ID.
      - Un utilisateur peut avoir max 5 tickets ouverts en même temps.

    Middleware

    - index.ts : Ajout de tous les middleware dans ce fichier pour facilite l'import.
    - auth.js : Ajout du middleware requireAdmin qui verifie que l'utilisateur est connecte et possede les bons droits admin.

    Modeles

    - index.ts : Ajout de tous les modeles dans ce fichier pour facilite l'import.
    - tickets.js : Ajout du modele Ticket qui correspond a celui de la base de donnees.
    - category.js : Creation du modele Category
    - user.js : Ajout de l'attribut ability dans le User ce qui permet d'etre admin des tickets et d'avoir les droits de :
      - visionner les tickets ouverts.
      - supprimer les tickets.

    Autre

    - app.js : Ajout de la route contact et category dans le routeur backend.

Tests Backend - cypress.config.ts : Ajout d'un parametre pour que les tests prennent des videos. - ticket.test.js : Creation de test pour la route /api/tickets. - Il y a aussi des petits tests en rapport avec les categories.
Pour effectuer les tests backend, il faut utiliser la commande : npm run test

FRONTEND

Modification Frontend

    Composants

    - index.ts : Ajout de tous les composants dans ce fichier pour facilite l'import.
    - Ticket.tsx : Creation du composant Ticket qui permet de l'utiliser plus facilement plusieurs fois.
      - Peut avoir l'attribut readonly pour ne pas avoir de boutons.
    - Navbar.tsx : Ajout d'un bouton pour acceder a la nouvelle page contact.
    - UserProvider.tsx : Ajout du provider pour concentrer toutes la variables "globales" (et inscrites dans le local storage) du site.
    - UserContext.tsx : Ajout du UserContext qui permet d'acceder au UserProvider.
    - Navbar.tsx : Ajout de liens vers les nouvelles pages et changement de navbar en fonction des droits et de l'etat de l'utilisateur.
    - components/Popup.tsx : Ajout d'un popup pour afficher a l'utilisateur qu'il a bien envoye son ticket
    - CategoryCard.tsx : Ajout de la carte Categorie pour plus de facilite lors de son utilisation multiple.

    Pages

    - index.ts : Ajout de toutes les pages dans ce fichier pour facilite l'import.
    - About.tsx : Ajout de la navbar dans la page.
    - CategoryManager : Creation de la page admin pour gerer les categories.
      - La page ne s'affiche seulement lorsque les droits de l'utilisateur ont ete checke auparavant.
    - ContactUs.tsx : Creation de la page contact.
      - Condition avec les inputs et toast error si ces tests ne sont pas valides.
      - Pour acceder a cette page, il faut cliquer sur la bouton dans la navbar ou aller sur LeNomDuSite/contact.
    - Dashboard.tsx : Ajout d'un lien vers la page contact car pas de navbar utilisable.
    - Login.tsx : Ajout de la reponse lors du login dans le local storage pour sauvegarder certains donnees insensibles. (Les donnees sauvegardee dans le local storage ne sont pas utilisees pour les requetes mais seulement afficher plus facilement certaines parties du frontend)
      - Ajout d'un toast pour signaler une erreur lors d'une redirection forcee a l'utilisateur.
    - TicketsViewer.tsx : Creation de la page admin pour manager les tickets (suppression, archivage et prise de contact avec l'utilisateur par mail)
      - La page ne s'affiche seulement si les droits ont ete checke auparavant.
      - Lors de la suppression ou le changement d'etat d'un ticket, la page est mise a jour sans etre rafraichie.
      - Details lors de la suppression d'un ticket.
      - Pas de pagination pour cette page. S'il y a beaucoup de tickets a charger d'un seul coup, peut poser problemes.
      - Bouton qui redirige vers le manager des categories exlusivement reserve aussi aux admins.
    - UserTicketsManager.tsx : Page utilisée par l'utilisateur pour voirs ses tickets ouverts. Les tickets dans cette pages sont en readonly.

    Types

    Ce dosser contient toutes les interfaces utilisees par certains composants et variables
    - index.ts : Ajout de tous les types dans ce fichier pour facilite l'import.

    Autres

    - app.tsx : Ajout de la page Contact, TicketViewer et CategoryManager dans le router du site.
    - main.tsx : Ajout du UserProvider dans l'application pour le mettre accessible a toutes la pages.

Tests Frontend

    Tests End-To-End

    - contact.cy.ts : Contient tous les tests e2e pour la page contact
      - Soumission du formulaire avec un probleme dans le nom de l'utilisateur.
      - Soumission du formulaire sans erreurs.
      - Soumission du formulaire avec des cases vides.

    Pour effectuer les tests unitaire il faut utiliser la commande : npm run test.e2e

    Tests unitaires

    - App.test.tsx : Ajout du test de l'app dans le dossier qui sera utilise lors de la commande.

    Pour effectuer les tests unitaire il faut utiliser la commande : npm run test.unit

Martin-Hayot committed:
Revert "US17"

This reverts commit 34bd81e912d765e126cf48cd38324fc3f2c97d7a.

Martin-Hayot committed:
refactor: navbar pour mobile et responsive

Samaze123 committed:
# US17

US perso de Samuel Lambert.
Creation de la page contact. Cette page est necessaire pour que l'utilisateur qui le souhaite puisse prendre contact avec l'equipe de la sonde.
Lors du contact avec l'equipe, le message de l'utilisateur est sauve dans un ticket.
Pour que l'equipe puisse lire et manager ces tickets, il a fallu creer une page, uniquement pour l'equipe.
Ensuite, pour plus de classification de ces tickets j'ai cree les categories qui sont manager exclusivement par l'equipe mais qui sont utilisable par l'utilisateur.

## BACKEND

### Modifications Backend

#### Routes

- index.ts : Ajout de toutes les routes dans ce fichier pour facilite l'import
user.js : Ajout d'une route pour recevoir l'ability d'un utilisateur.
- auth.js : Ajout d'une reponse avec des donnees de l'utilisateur qui seront stockee dans le localstorage du front. Ajout de ability lors de la creation d'un compte.

- category.js : Creation de la route pour ajouter, supprimer et recevoir les categories.
  - Il y a aussi une route qui sert a recevoir seulement une categorie grace a son ID.

- ticket.js : Creation de la route pour ajouter, supprimer et recevoir les tickets.
  - Utilisation du middleware RequireAdmin utilise pou certaines url.
  - Details lors du retour si mauvaises donnees entrees.
  - Il y a aussi une route qui sert a recevoir seulement un ticket grace a son ID.

#### Middleware

- index.ts : Ajout de tous les middleware dans ce fichier pour facilite l'import.
- auth.js : Ajout du middleware requireAdmin qui verifie que l'utilisateur est connecte et possede les bons droits admin.

#### Modeles

- index.ts : Ajout de tous les modeles dans ce fichier pour facilite l'import.
- tickets.js : Ajout du modele Ticket qui correspond a celui de la base de donnees.
- category.js : Creation du modele Category
- user.js : Ajout de l'attribut ability dans le User ce qui permet d'etre admin des tickets et d'avoir les droits de :
  - visionner les tickets ouverts.
  - supprimer les tickets.

#### Autre

- app.js : Ajout de la route contact et category dans le routeur backend.

### Tests Backend
- cypress.config.ts : Ajout d'un parametre pour que les tests prennent des videos.
- ticket.test.js : Creation de test pour la route /api/tickets.
  - Il y a aussi des petits tests en rapport avec les categories.
  Pour effectuer les tests backend, il faut utiliser la commande : npm run test

## FRONTEND

### Modification Frontend

#### Composants

- index.ts : Ajout de tous les composants dans ce fichier pour facilite l'import.
- Ticket.tsx : Creation du composant Ticket qui permet de l'utiliser plus facilement plusieurs fois.
- Navbar.tsx : Ajout d'un bouton pour acceder a la nouvelle page contact.
- UserProvider.tsx : Ajout du provider pour concentrer toutes la variables "globales" (et inscrites dans le local storage) du site.
- UserContext.tsx : Ajout du UserContext qui permet d'acceder au UserProvider.
- Navbar.tsx : Ajout de liens vers les nouvelles pages et changement de navbar en fonction des droits et de l'etat de l'utilisateur.
- components/Popup.tsx : Ajout d'un popup pour afficher a l'utilisateur qu'il a bien envoye son ticket
- CategoryCard.tsx : Ajout de la carte Categorie pour plus de facilite lors de son utilisation multiple.

#### Pages

- index.ts : Ajout de toutes les pages dans ce fichier pour facilite l'import.
- About.tsx : Ajout de la navbar dans la page.
- CategoryManager : Creation de la page admin pour gerer les categories.
  - La page ne s'affiche seulement lorsque les droits de l'utilisateur ont ete checke auparavant.
- ContactUs.tsx : Creation de la page contact.
  - Condition avec les inputs et toast error si ces tests ne sont pas valides.
  - Pour acceder a cette page, il faut cliquer sur la bouton dans la navbar ou aller sur LeNomDuSite/contact.
- Dashboard.tsx : Ajout d'un lien vers la page contact car pas de navbar utilisable.
- Login.tsx : Ajout de la reponse lors du login dans le local storage pour sauvegarder certains donnees insensibles. (Les donnees sauvegardee dans le local storage ne sont pas utilisees pour les requetes mais seulement afficher plus facilement certaines parties du frontend)
  - Ajout d'un toast pour signaler une erreur lors d'une redirection forcee a l'utilisateur.
- TicketsViewer.tsx : Creation de la page admin pour manager les tickets (suppression, archivage et prise de contact avec l'utilisateur par mail)
  - La page ne s'affiche seulement si les droits ont ete checke auparavant.
  - Lors de la suppression ou le changement d'etat d'un ticket, la page est mise a jour sans etre rafraichie.
  - Details lors de la suppression d'un ticket.
  - Pas de pagination pour cette page. S'il y a beaucoup de tickets a charger d'un seul coup, peut poser problemes.
  - Bouton qui redirige vers le manager des categories exlusivement reserve aussi aux admins.

#### Types

Ce dosser contient toutes les interfaces utilisees par certains composants et variables
- index.ts : Ajout de tous les types dans ce fichier pour facilite l'import.

#### Autres

- app.tsx : Ajout de la page Contact, TicketViewer et CategoryManager dans le router du site.
- main.tsx : Ajout du UserProvider dans l'application pour le mettre accessible a toutes la pages.

### Tests Frontend

#### Tests End-To-End

- contact.cy.ts : Contient tous les tests e2e pour la page contact
  - Soumission du formulaire avec un probleme dans le nom de l'utilisateur.
  - Soumission du formulaire sans erreurs.
  - Soumission du formulaire avec des cases vides.

Pour effectuer les tests unitaire il faut utiliser la commande : npm run test.e2e

#### Tests unitaires

- App.test.tsx : Ajout du test de l'app dans le dossier qui sera utilise lors de la commande.

Pour effectuer les tests unitaire il faut utiliser la commande : npm run test.unit

Kouvahe81 committed:
installation module et import

Kouvahe81 committed:
fix:correction

Kouvahe81 committed:
Merge branch 'feature/US7-2/Signup-Create-new-user' of https://github.com/Martin-Hayot/Projet-Integration-2023 into feature/US7-2/Signup-Create-new-user

JarodEphec reviewed (changes requested):
Je vois qu'il y a des commits qui ne sont pas à toi dans ta pull request, ça m'effraie beaucoup.
Quand le problème sera réglé, je rééprouverai ta pull request.

Julien00859 commented:
Doucement sur ce genre de commentaire, ils peuvent être mal pris. Au lieu de bloquer la PR, ne serait-il pas mieux que vous essayez de corriger le problème ensemble. La branche a plus d'un mois d'existance, il est plus que temps de la merger, tout devrait être mis en oeuvre pour débloquer la situation. Si je devais utiliser terminologie Kanban, je dirais qu'il est temps de "pull the trigger": on arrête tout et on se concentre tous sur venir en aide à notre collègue. Vous pouvez venir me trouver sur discord si vous voulez qu'on mette de l'ordre dans cet historique.

Martin-Hayot commented:
Suite au problème d'historique Serge à créer une nouvelle pull request qui reprends son travail fait ici. 
Je ferme donc cet pull request.

JarodEphec commented:
La nouvelle pull request est la #33 .
